cypress/integration/basics.js

```diff
@@ -216,4 +265,71 @@ describe('JSON Import', function() {
     });
 
+    it('works for new codemeta v3.0 terms', function() {
+        cy.get('#codemetaText').then((elem) =>
+            elem.text(JSON.stringify({
+                "@context": "https://w3id.org/codemeta/3.0",
+                "type": "SoftwareSourceCode",
+                "name": "My Test Software",
+                "continuousIntegration": "https://test-ci.org/my-software",
+                "isSourceCodeOf": "Bigger Application",
+                "review": {
+                    "type": "Review",
+                    "reviewAspect": "Some software aspect",
+                    "reviewBody": "Some review"
+                }
+            }))
+        );
+        cy.get('#importCodemeta').click();
+
+        cy.get('#contIntegration').should('have.value', 'https://test-ci.org/my-software');
+        cy.get('#isSourceCodeOf').should('have.value', 'Bigger Application');
+        cy.get('#reviewAspect').should('have.value', 'Some software aspect');
+        cy.get('#reviewBody').should('have.value', 'Some review');
+    });
+
+    it('works for codemeta v2.0 terms in v3.0 version', function() {
+        cy.get('#codemetaText').then((elem) =>
+            elem.text(JSON.stringify({
+                "@context": "https://w3id.org/codemeta/3.0",
+                "type": "SoftwareSourceCode",
+                "name": "My Test Software",
+                "continuousIntegration": "https://test-ci.org/my-software",
+                "codemeta:contIntegration": {
+                    "id": "https://test-ci.org/my-software"
+                },
+            }))
+        );
+        cy.get('#importCodemeta').click();
+
+        cy.get('#contIntegration').should('have.value', 'https://test-ci.org/my-software');
+    });
+
+    it('works for codemeta v3.0 terms in v2.0 version, and does not work for new terms', function() {
+        cy.get('#codemetaText').then((elem) =>
+            elem.text(JSON.stringify({
+                "@context": "https://doi.org/10.5063/schema/codemeta-2.0",
+                "type": "SoftwareSourceCode",
+                "name": "My Test Software",
+                "contIntegration": "https://test-ci.org/my-software",
+                "codemeta:continuousIntegration": {
+                    "id": "https://test-ci.org/my-software"
+                },
+                "codemeta:isSourceCodeOf": {
+                    "id": "Bigger Application"
+                },
+                "schema:review": {
+                    "type": "schema:Review",
+                    "schema:reviewAspect": "Some software aspect",
+                    "schema:reviewBody": "Some review"
+                }
+            }))
+        );
+        cy.get('#importCodemeta').click();
+
+        cy.get('#contIntegration').should('have.value', 'https://test-ci.org/my-software');
+        cy.get('#isSourceCodeOf').should('have.value', '');
+        cy.get('#reviewAspect').should('have.value', '');
+        cy.get('#reviewBody').should('have.value', '');
+    });
```

Comment:
The context should not impact the semantics of the document, it's only for human-readability.

> Should we always import using the latest version (v3.0 in this case)?

If only the v3 term is given, then yes.

If both are given but have a different value, then that's a good question. Semantically, both the old and the new field are equivalent so it doesn't matter which one we pick. I guess users would expect the one "with the context's version" to have precedence if both are given?

Reply:
And I'm not sure how to address the latter case. Should we use the context to determine the version for automatic generation ?

Reply:
So, in the end, the one that is picked is the one with the newest term (e.g. between `codemeta:contIntegration` and `continuousIntegration` or `contIntegration` and `codemeta:continuousIntegration`, it will always import `[codemeta:]continuousIntegration`).

> It also raises the question of automatically generating v2.0 by default. I'm seeing now that it can introduce some weird behavior to the user if he tries to import a codemeta v3.0 document, then update the form, and then sees the output as a codemeta v2.0 document.

And the update linked to this comment also resolves this question, as the generated document now always contains all properties, no matter which version they're from, which should not disturb the user (the properties will always appear, with or without prefix).